src/components/SearchFilter/SearchFilter.js

```diff
@@ -0,0 +1,39 @@
+import { useCallback, useEffect, useState } from "react";
+import { getData } from "../../services/api";
+
+//can be moved inside the SearchFilter component;
+function debounced(debounceFunc, timeout) {
+  let timer;
+  return (...args) => {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+      debounceFunc.apply(this, args);
+    }, timeout);
+  };
+}
+
+const SearchFilter = ({ setUsers }) => {
+  const [searchTerm, setSearchTerm] = useState("");
+
+  async function loadFiltredUsers(searchTerm) {
+    let response = await getData(searchTerm);
+    setUsers(response.data);
+  }
+
+  const debounce = useCallback(debounced(loadFiltredUsers, 300), []);
```

Comment:
why is debounce in the dependency array

Reply:
because it throws warning of missing dependency